### PR TITLE
chore: automate release of python client

### PR DESCRIPTION
**Pull request by Bento007**
# Reason
https://github.com/chanzuckerberg/cryoet-data-portal/issues/902
- This stream lines the release process and reduces the overhead required to release.

# Changes
- [fix: add link to documentation to pypi](https://github.com/chanzuckerberg/cryoet-data-portal/pull/972/commits/38c76b450294d35a276d102ed816b12dd810ee99)
- [chore: prepare client for release automation](https://github.com/chanzuckerberg/cryoet-data-portal/pull/972/commits/d7e5446519aebfbc180eaef98a11ce65ec7d49dc)
  - add CHANGELOG.md for release-please
  - add make build recipe for building the pypi release
  - hardcode version in client
  - add client to release-please config and manifest
  - add update py-build.yml to use release please.
- chore: automate build and release of pypi package after release PR is merged.
- update release documentation for client using release-please

# Testing
- generated [release PR ](https://github.com/chanzuckerberg/cryoet-data-portal/pull/981)
- testpypi https://github.com/chanzuckerberg/cryoet-data-portal/actions/runs/10187905818/job/28182966923?pr=972
- [testpypi release](https://test.pypi.org/project/cryoet-data-portal/)

 # Notes
Moving forwards all commits to the client that should be represented in the client CHANGELOGS.md must be written using the conventional commit format.

### CODE REVIEW

File: .github/workflows/py-build.yml
Comment (manasaV3) [inline .github/workflows/py-build.yml:6]: We shouldn't publish for each push to main. We should have this workflow be runnable on dispatch, and should limit the people who can allow this to be executed.

File: .github/workflows/py-build.yml
Comment (manasaV3) [inline .github/workflows/py-build.yml:38]: This might need to be updated.. :)

File: release-please.config.json
Comment (manasaV3) [inline release-please.config.json:2]: Curious on why we are removing this?

**Review by manasaV3 — CHANGES_REQUESTED** [review]

File: release-please.config.json
Comment (Bento007) [reply release-please.config.json:2]: It's only used the first time release-please runs. So it is now a noop. [Docs](https://github.com/googleapis/release-please/blob/ace2bd5dc778f83c33ad5dee6807db5d0afdba36/docs/manifest-releaser.md?plain=1#L132)

File: .github/workflows/py-build.yml
Comment (Bento007) [reply .github/workflows/py-build.yml:6]: > We shouldn't publish for each push to main.

The release of the API is gates by https://github.com/chanzuckerberg/cryoet-data-portal/blob/8fcc103fceebcfd18c5b94f3ee14baa8239f83d6/.github/workflows/py-build.yml#L27-L27
Which is only true when a release PR is merged into main for the client. So who ever has permission to merge a PR has permission to release the client.

>  limit the people who can allow this to be executed

Who should the release be limited to? As the PR is currently written, you need permission to merge a PR to release the client.

**Comment by Bento007** [comment]
> overall lgtm. how do you plan on merging the release PR? is that something you'll be doing manually?

That's the workflow i'm moving towards if @manasaV3 agrees.

**Comment by Bento007** [comment]
>run release-please in a single workflow to avoid any unexpected concurrency issues. This involved combining [py-build.yml](https://github.com/chanzuckerberg/cryoet-data-portal/blob/main/.github/workflows/py-build.yml) and [create-frontend-release-pr.yml](https://github.com/chanzuckerberg/cryoet-data-portal/blob/main/.github/workflows/create-frontend-release-pr.yml) into a single workflow that runs on push to main.

I went ahead and did this because it would likely cause race conditions with release-please if we didn't do it now.

File: .github/workflows/release.yml
Comment (Bento007) [reply .github/workflows/release.yml:29]: Yes is should. This was for testing. I'll change it before i merge.

Comment (Bento007) [reply .github/workflows/release.yml:28]:
```suggestion
          target-branch: "main"
```

File: .github/workflows/release.yml
Comment (jgadling) [inline .github/workflows/release.yml:7]: We need a filter on this to only run when the client directory changes, I think?

File: client/python/cryoet_data_portal/pyproject.toml
Comment (jgadling) [inline client/python/cryoet_data_portal/pyproject.toml:7]: Does release please update this?

Comment (manasaV3) [reply client/python/cryoet_data_portal/pyproject.toml:7]: And the [src/cryoet_data_portal/__init__.py](https://github.com/chanzuckerberg/cryoet-data-portal/pull/972/files#diff-51bfa7b6ff3f0643c80e954b9686c748cf90b433b5e5822ab4b2cd0cf41d04a0) file.

File: .github/workflows/py-build.yml
Comment (manasaV3) [reply .github/workflows/py-build.yml]: > So who ever has permission to merge a PR has permission to release the client.

Currently all the contributors can merge PR. I think that should be fine as that limits unknown external actors from kicking of a deploy.

File: client/python/cryoet_data_portal/pyproject.toml
Comment (Bento007) [reply client/python/cryoet_data_portal/pyproject.toml:7]: yes, it updates both

File: .github/workflows/release.yml
Comment (Bento007) [reply .github/workflows/release.yml:7]: That is now configured in `release-please.config.json`. Every time we merge release-please will run and check if any committed changes affect those paths and update the release pr as needed.

**Review by manasaV3 — APPROVED** [review]
🚢

File: .github/workflows/release.yml
Comment (jgadling) [inline .github/workflows/release.yml:28]: Is this missing the token parameter from the frontend action?
```
token: ${{ secrets.GITHUB_TOKEN }}
```

**Review by jgadling — APPROVED** [review]
I suspect the token parameter is important for release please, but this lgtm once that's sorted out.

**Pull request merged**
